buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,391 @@
+@Library('dotnet-ci') _
+
+// Incoming parameters.  Access with "params.<param name>".
+// Note that the parameters will be set as env variables so we cannot use names that conflict
+// with the engineering system parameter names.
+
+//--------------------- Windows Functions ----------------------------//
+
+def windowsBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoStash = ((pgo == "skiprestoreoptdata") ? "nopgo" : "pgo")
+    String baselineString = ""
+
+    // For baseline builds, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        bat "git checkout HEAD^^1"
+    }
+
+    bat "set __TestIntermediateDir=int&&.\\build.cmd ${config} ${arch} skipbuildpackages ${pgo}"
+
+    // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
+    stash name: "nt-${arch}-${pgoStash}${baselineString}-build-artifacts", includes: 'bin/**'
+    stash name: "nt-${arch}-${pgoStash}${baselineString}-test-artifacts", includes: 'bin/tests/**'
```

Comment:
Just so that I understand, this is just the tests, and the one above stashes the tests and the binaries, yes?

Reply:
Correct. I could potentially only stash the bin directory, and pull that in on linux too, but I don't know if that's necessary.